### PR TITLE
UAR-617: add BO and MO private data retrieval from API

### PR DESCRIPTION
**Pull request by n-r-b**
### JIRA link
UAR-617

### Change description

Add BO and MO private data retrieval functions.

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/service/private.overseas.entity.details.ts [hunk 1]
```diff
@@ -18,15 +22,59 @@ export const getPrivateOeDetails = async (
...
+const checkErrorResponse = (req: Request, response, overseasEntityId?: string, transactionId?: string, dataToRetrieve: string = "") => {
   if (response.httpStatusCode !== 200 && response.httpStatusCode !== 404) {
-    const errorMsg = `Something went wrong fetching overseas entity details = ${JSON.stringify(response)}`;
+    const errorMsg = `Something went wrong fetching private ${dataToRetrieve} details = ${JSON.stringify(response)}`;
     throw createAndLogErrorRequest(req, errorMsg);
   }
 
   if (response.httpStatusCode === 404) {
-    logger.debugRequest(req, `No overseas entity details found ${overseasEntityId} under ${transactionId}`);
+    logger.debugRequest(req, `No private ${dataToRetrieve} found for ${overseasEntityId} under ${transactionId}`);
```
Comment (AndreKramer7654) [inline src/service/private.overseas.entity.details.ts:75]: add "details" after ${}

File: src/service/private.overseas.entity.details.ts [hunk 2]
```diff
@@ -18,15 +22,59 @@ export const getPrivateOeDetails = async (
...
     return undefined;
   }
 
-  return response.resource;
+  logger.debugRequest(req, `${response}`);
```
Comment (AndreKramer7654) [inline src/service/private.overseas.entity.details.ts:79]: not all code path return value now

**Review by bbell-companieshouse — APPROVED** [review]

File: src/service/private.overseas.entity.details.ts (hunk 2 above)
Comment (AndreKramer7654) [reply src/service/private.overseas.entity.details.ts:79]: remove debug log

Comment (n-r-b) [reply src/service/private.overseas.entity.details.ts]: Removed

File: src/service/private.overseas.entity.details.ts (hunk 1 above)
Comment (n-r-b) [reply src/service/private.overseas.entity.details.ts]: Done

**Review by AndreKramer7654 — APPROVED** [review]

**Pull request merged**